src/types/channels.ts

```diff
@@ -73,4 +73,27 @@ export type ChannelsState = {
     groupsAssociatedToChannel: any;
     totalCount: number;
     manuallyUnread: RelationOneToOne<Channel, boolean>;
+    channelModerations: RelationOneToOne<Channel, Array<ChannelModeration>>;
+};
+
+export type ChannelModeration = {
+    name: string;
+    roles: {
+        guests?: {
+            value: boolean;
+            enabled: boolean;
+        };
+        members: {
```

Comment:
```suggestion
        members?: {
```

Reply:
I think a moderation will always have a member attribute - since all channel moderations apply to the members role but not necessarily to guests (manage members permission)